sdks/python/apache_beam/testing/datatype_inference_test.py

```diff
@@ -174,6 +175,7 @@ def test_infer_typehints_schema(self, _, data, schema):
   @parameterized.expand([(d["name"], d["data"], d["pyarrow_schema"])
                          for d in TEST_DATA])
   @unittest.skipIf(pa is None, "PyArrow is not installed")
+  @unittest.skipIf(sys.platform == "win32", "[BEAM-10624]")
```

Comment:
@TheNeuralBit - - assigned this jira to you.

Reply:
Ack I'll take a look. It may just take a pyarrow upgrade, which will be unblocked once we drop python 2

Reply:
Looks like it's not actually related to pyarrow version. I think we should fix it by making the dtype explicit for the numpy arrays created here: https://github.com/apache/beam/blob/8d03dc698dcea1745a26c9823c8d942f5634c6b1/sdks/python/apache_beam/testing/datatype_inference_test.py#L57

I have a personal Windows machine I could clone Beam on and verify, but it might be simpler to just test the fix as part of this PR.

Reply:
@TheNeuralBit could take a look on https://github.com/apache/beam/pull/12452/commits/4004309b2b32acd1451cd30e92ab14f645a51211 , please?
Is it the solution you mentioned?
I tested it on my local fork (run resulsts [here](https://github.com/TobKed/beam/actions/runs/199604657)) and it seems to work.

Reply:
Yep that's what I was suggesting!

Reply:
Thanks @TheNeuralBit.  I cherry-picked change to the PR.